### PR TITLE
MG-18 - Add cookie expiration time

### LINKED ISSUE #18: [FEATURE] Add Token Expiration time to token cookies
### Is there an existing feature request for this?

- [X] I have searched the existing feature requests

### Is your feature request related to a problem? Please describe.

_No response_

### Describe the feature you are requesting, as well as the possible use case(s) for it.

We add expiration time to tokens as we add them to browser cookies

### Indicate the importance of this feature to you.

- [ ] Must-have
- [X] Should-have
- [ ] Nice-to-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ianmuchyri**
# What type of PR is this?

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Dependency Update

## What does this do?
Add the ability to set token expiration time from the `.env` file. These values are then added to the cookie `max-age` variable

## Which issue(s) does this PR fix/relate to?

- Related Issue #
- Resolves #18 

## Have you included tests for your changes?

- [ ] Yes
- [x] No, and this is why: 

## Did you document any new/modified functionality?

- [ ] Yes
- [x] No, and this is why:

### CODE REVIEW

**Comment by fbugarski** [comment]
LGTM

File: docker/.env [hunk 1]
```diff
@@ -17,6 +17,8 @@ MG_UI_HOST_URL=http://localhost:9095
 MG_UI_VERIFICATION_TLS=false
 MG_UI_CONTENT_TYPE = application/senml+json
 MG_UI_INSTANCE_ID=
+MG_ACCESS_TOKEN_TTL=24h
+MG_REFRESH_TOKEN_TTL=24h
```
Comment (rodneyosodo) [inline docker/.env:21]: Why do we need this and UI is not generating any token? Shouldn't this information be decoded from the token itself?

File: ui/api/endpoint.go [hunk 2]
```diff
@@ -260,18 +286,21 @@ func refreshTokenEndpoint(svc ui.Service) endpoint.Endpoint {
 					Value:    token.AccessToken,
 					Path:     "/",
 					HttpOnly: true,
+					MaxAge:   exp,
```
Comment (rodneyosodo) [inline ui/api/endpoint.go:289]: Why not use the `Expires` field since we have the time when the token expires rather than subtracting `time.Now` ?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: ui/api/endpoint.go [hunk 3]
```diff
@@ -225,12 +245,14 @@ func tokenEndpoint(svc ui.Service) endpoint.Endpoint {
 					Value:    token.AccessToken,
 					Path:     "/",
 					HttpOnly: true,
+					Expires:  exp,
```
Comment (rodneyosodo) [inline ui/api/endpoint.go:248]: This expiry time for refresh token but not access token

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: ui/api/endpoint.go (hunk 3 above)
Comment (ianmuchyri) [reply ui/api/endpoint.go:248]: yes, I explained the reason behind my doing this on the UI channel. Check the conversation we had with Dusan

Comment (ianmuchyri) [reply ui/api/endpoint.go]: This was resolved based on this explanation:

```
For access_token with 15min

- if token is non-existent and refresh_token is there use refresh_token to fetch new token i.e will not send you to login page
- if token is non-existent and refresh_token is non-existent send user to login page
```

Comment (dborovcanin) [reply ui/api/endpoint.go]: https://github.com/absmach/magistrala-ui/issues/18#issuecomment-1906001123

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**